web/src/message_edit.ts

```diff
@@ -926,25 +996,26 @@ export function do_save_inline_topic_edit($row, message, new_topic) {
     });
 }
 
-export function save_message_row_edit($row) {
+export function save_message_row_edit($row: JQuery): void {
     assert(message_lists.current !== undefined);
     const $banner_container = compose_banner.get_compose_banner_container(
         $row.find(".message_edit_form textarea"),
     );
     const stream_id = Number.parseInt(
-        rows.get_message_recipient_header($row).attr("data-stream-id"),
+        rows.get_message_recipient_header($row).attr("data-stream-id")!,
         10,
     );
     const msg_list = message_lists.current;
     let message_id = rows.id($row);
     let message = message_lists.current.get(message_id);
+    assert(message !== undefined);
     let changed = false;
     let edit_locally_echoed = false;
 
     let new_content;
     const old_content = message.raw_content;
 
-    const $edit_content_input = $row.find(".message_edit_content");
+    const $edit_content_input = $row.find<HTMLInputElement>(".message_edit_content");
```

Comment:
This is a `<textarea>`, not an `<input>`. Also adjust the selector string to guarantee type safety.

```suggestion
    const $edit_content_input = $row.find<HTMLTextAreaElement>("textarea.message_edit_content");
```